ironfish-cli/src/commands/wallet/mint.ts

```diff
@@ -142,42 +160,116 @@ export class Mint extends IronfishCommand {
     }
 
     let fee
-    if (flags.fee) {
-      fee = flags.fee
-    } else {
-      const input = await CliUx.ux.prompt(
-        `Enter the fee amount in $IRON (min: ${CurrencyUtils.renderIron(1n)})`,
-        {
-          default: CurrencyUtils.renderIron(1n),
-          required: true,
-        },
-      )
+    let rawTransactionResponse: string
 
-      fee = await parseIron(input, { largerThan: 0n, flagName: 'fee' }).catch((error: Error) =>
-        this.error(error.message),
-      )
+    try {
+      if (flags.fee) {
+        fee = flags.fee
+
+        const mintResponse = await client.mintAsset({
+          account,
+          assetId,
+          fee: CurrencyUtils.encode(fee),
+          metadata,
+          name,
+          value: CurrencyUtils.encode(amount),
+          confirmations,
+        })
+
+        rawTransactionResponse = mintResponse.content.transaction
+      } else {
+        const feeRatesResponse = await client.estimateFeeRates()
+        const feeRates = new Set([
+          feeRatesResponse.content.low ?? '1',
+          feeRatesResponse.content.medium ?? '1',
+          feeRatesResponse.content.high ?? '1',
+        ])
+
+        const feeRateNames = Object.getOwnPropertyNames(feeRatesResponse.content)
+
+        const feeRateOptions: { value: number; name: string }[] = []
+
+        const mintRequest: MintAssetRequest = {
+          account,
+          assetId,
+          metadata,
+          name,
+          value: CurrencyUtils.encode(amount),
+          confirmations,
+        }
+
+        const allPromises: Promise<RpcResponseEnded<MintAssetResponse>>[] = []
+        feeRates.forEach((feeRate) => {
+          allPromises.push(
+            client.mintAsset({
+              ...mintRequest,
+              feeRate: feeRate,
+            }),
+          )
+        })
+
+        const mintResponses = await Promise.all(allPromises)
+        mintResponses.forEach((mintResponse, index) => {
+          const rawTransactionBytes = Buffer.from(mintResponse.content.transaction, 'hex')
+          const rawTransaction = RawTransactionSerde.deserialize(rawTransactionBytes)
+
+          feeRateOptions.push({
+            value: index,
+            name: `${feeRateNames[index]}: ${CurrencyUtils.renderIron(
+              rawTransaction.fee,
+            )} IRON`,
+          })
+        })
+
+        const input: { selection: number } = await inquirer.prompt<{ selection: number }>([
+          {
+            name: 'selection',
+            message: `Select the fee you wish to use for this transaction`,
+            type: 'list',
+            choices: feeRateOptions,
+          },
+        ])
+
+        rawTransactionResponse = mintResponses[input.selection].content.transaction
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        this.error(error.message)
+      }
+      throw error
```

Comment:
You can delete the try catch it already exists in IronfishCommand and does this logic.